wagtail/snippets/static_src/wagtailsnippets/js/snippet-chooser.js

```diff
@@ -4,17 +4,21 @@ function createSnippetChooser(id, modelString) {
     var input = $('#' + id);
     var editLink = chooserElement.find('.edit-link');
 
+    function snippetChosen(snippetData, initial) {
+        if (!initial) {
```

Comment:
Every time one of the fields in a StreamField gets modified, a boolean gets switched to `true` that will show a confirm dialog before leaving the page. So this check was done to prevent the page to be considered as modified when loading it. In addition to that, the value is already set in the `input` when rendering the template, so it does not need to be set again.